test/lib/appFactory.ts

```diff
@@ -0,0 +1,52 @@
+import { SNApplication, Environment, Platform, SNLog, SNComponentManager } from '@standardnotes/snjs';
+import DeviceInterface from './deviceInterface';
+import SNCrypto from './snCrypto';
+import { WebComponentManager, MobileComponentManager } from './componentManager';
+
+const getSwappedClasses = (environment: Environment) => {
```

Comment:
same here - this should be a class working under single responsibility principle instead of a bunch of functions in a file.

Reply:
There is only one exported function in this file which communicates clearly where its boundaries are, add to that the fact that there is no state to manage and that makes using a class not needed imo

Reply:
I don't agree - this is not OO design and leads to code that is hard to test and maintain. Also it becomes unclear how classes depend on each other and where is the complexity.

For example: let's say we have a class `Foo` and a method in this class `Foo:bar` uses this function directly. How will you write tests for `Foo:bar` that will take into consideration different outcomes of `getSwappedClasses`. Some hacky mock techniques have to be used to globally stub this function. Whereas if we wrapped in the a single responsibility class we could clearly separate concerns of this code and isolate tests by mocking dependencies of a given class.

Real life, simple example 1:
- class: https://github.com/standardnotes/syncing-server-js/blob/develop/src/Domain/Item/ContentDecoder.ts
- tests: https://github.com/standardnotes/syncing-server-js/blob/develop/src/Domain/Item/ContentDecoder.spec.ts

Real life example with other classes as dependencies:
- class: https://github.com/standardnotes/syncing-server-js/blob/develop/src/Domain/UseCase/ClearLoginAttempts.ts
- tests: https://github.com/standardnotes/syncing-server-js/blob/develop/src/Domain/UseCase/ClearLoginAttempts.spec.ts

Clean code should follow `S.O.L.I.D` principles.

Reply:
@karolsojko this file is not part of the actual library code. Everything under `test/lib` is used only to create the `SNApplication` instance, which is needed to test component actions. It would be something similar to what `snjs` has here https://github.com/standardnotes/snjs/tree/master/test/lib

Reply:
my bad - for test purposes it's ok